### DISCUSSION TITLE
Broken RichTextView style modification

### DISCUSSION THREAD

**Discussion by IvanStebletsov** [danielsaidi/RichTextKit#207] [Q&A]
I wanted to change a `RichTextEditor` style, applied `richTextEditorStyle` modificator and noticed that it works only if the text field is empty. In case it has something, modificator does nothing. For example, I run `Demo App` with applied `richTextEditorStyle`, style changed as expected, I type something, restart app and again the style changed to default. 

I conducted small investigation and came across this strange line in `setup` method of `RichTextView`:

```swift
extension RichTextView {
    ...

    func setup(_ theme: RichTextView.Theme) {
        guard richText.string.isEmpty else { return }  // <---  It doesn't allow me to change TextEditor style
        font = theme.font
        textColor = theme.fontColor
        backgroundColor = theme.backgroundColor
    }
}
```

As I understood this way you wanted to avoid extra UI updates?

https://github.com/user-attachments/assets/a4c55eea-1055-45c1-8574-bb1883b72421

**Comment by danielsaidi** [comment]
Great find @IvanStebletsov 🙌
